### PR TITLE
Feature/enhance build process

### PR DESCRIPTION
**Pull request by miton18**
- wait-for-it
- badges
- CRLF to LF
Don't copy node_modules into docker (yarn install it)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,52 +1,53 @@
...
-    secure: Kn9btER/9HJLCoptbzCbJm4HR0m48M/XsXE8Kxh8yeTMQwV2HR4j6EiKZIx2KV1HZZhNKokDRoZay9NV61Q5JTrmWyDt2cw3GXHOtlblgV6qXWH4GdRhW6pL4zmkEsmfoZl4+g/0j6t1R2eCD1rSg43C9VEiwBAcd0u98CTOnmlbijopMIJuGSbFVZI1UX7fqqQ0GHVHymmHVEYcB4vFbQX3hUy1PSRvB9h1ZDoLP5ZajcDDyfezwWuufsdCxLsob1dojJuQkhxDmcnwLXLwn7wHepwW3AYVu3k26yiITT9DeTEU6mSxTaCFhbHhKz4G1jHYsAKle0OUwXVb49GOf75fcNILnGxbVteZ2Pczqm7bk4vA28FFmm40H40oTz8LFFbjg3bZX3ZLKTHutNGYEehWs1U5w6Dj7Ws/pc9DZUfVnCtuT6KBILKXg0flL0ZjZEC7Ydvi1h3FChrYyCO9IHGUdllfHroW39IBPjPT7C/FW3MMRSfCAzlQbxseq4AOTdQJ/jTe/vRzWEMjnIZM1AwPGp/4pJ6Z3GYPIWDDLQq+yTyt4CJxwgV8c6ZpyREGK+Dg7I4XiZHd1b0AiU3U+wtYLOj9cQHzWFNoF/vvoKh6jYzJK65pDdzvFeuedE/sQz9eHCJb8QWAR231oQXlJDXrhhOWygROC5Kt/z0HzYk=
-  on:
-    tags: true
-    all_branch: true
+sudo: required
+
+language: node_js
+
+node_js:
+- 7.2.1
```
Comment (FlorentinDUBOIS) [inline .travis.yml]: You should use node or just 7 instead

File: .travis.yml [hunk 2]
```diff
@@ -1,52 +1,53 @@
...
+services:
+- mongodb
+- docker
+
+cache:
+  directories:
+  - node_modules
+
+install:
+- npm install --only=dev
```
Comment (FlorentinDUBOIS) [inline .travis.yml]: Use yarn instead

File: Dockerfile [hunk 3]
```diff
@@ -1,16 +1,22 @@
 FROM node:7.2.1
```
Comment (FlorentinDUBOIS) [inline Dockerfile]: You should use node:latest or node:7 instead

File: Dockerfile [hunk 4]
```diff
@@ -1,16 +1,22 @@
...
+COPY yarn.lock /usr/src/web-proxy/
+RUN yarn install --prod
+
+# Copy app
+COPY . /usr/src/web-proxy
 
 EXPOSE 80 443 8080
 
-ENTRYPOINT [ "npm", "start" ]
+ENTRYPOINT npm start
```
Comment (FlorentinDUBOIS) [inline Dockerfile]: You should use yarn start instead

File: docker-compose.yml [hunk 5]
```diff
@@ -10,11 +10,11 @@ services:
...
-      PROXY_DB: "mongodb://web-proxy-mongo/proxy"
-      PROXY_KEY: "a"
-      PROXY_SALT: "a"
-      #PROXY_OVH_KEY: ""
-      #PROXY_TRACE_KEY: ""
+      PROXY_MONGODB_ADDON_URI: "mongodb://web-proxy-mongo:27017/proxy"
+      PROXY_JWT_SECRET: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+      PROXY_PEPPER: "a"
+      PROXY_JWT_ISSUER: "aaa"
+      PROXY_JWT_AUDIENCE: "bbbb"
```
Comment (FlorentinDUBOIS) [inline docker-compose.yml]: Should we use empty values instead ?

**Review by FlorentinDUBOIS — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (miton18) [reply .travis.yml]: Yes ! so I put "node" for the latest and v6.9.4 (last LTS version)

File: docker-compose.yml (hunk 5 above)
Comment (miton18) [reply docker-compose.yml]: App won't start if it's empty, but I agree with you, we can put better temporary passwords

**Review by FlorentinDUBOIS — APPROVED** [review]

**Pull request merged**